api/api.go

```diff
@@ -0,0 +1,77 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/NebulousLabs/skynet-accounts/database"
+
+	"github.com/NebulousLabs/skynet-accounts/build"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+// API is ...
+type API struct {
+	DB     *database.DB
+	Router *httprouter.Router
+}
+
+// New returns a new initialised API.
+func New() *API {
+	router := httprouter.New()
+	router.RedirectTrailingSlash = true
+
+	api := &API{
+		Router: router,
+	}
+	api.buildHTTPRoutes()
+
+	db, err := database.New(context.Background())
+	api.DB = db
+	if err != nil {
+		log.Println("Failed to connect to the database! Running without a database.")
+		// Assign an empty database instance. In its methods we can check for
+		// the availability of the DB and return errors instead of crashing.
+		api.DB = &database.DB{}
```

Comment:
My idea was that for people who want to run a portal but don't want to run premium accounts or any kind of accounts, MongoDB won't be strictly necessary. This could be someone who's running a personal Skynet portal or an internal Skynet portal for their company/team. If they don't want to limit their users it's perfectly fine to forego Mongo and run with it.

Thinking about it again, those people won't really need to run `user-accounts` at all. I'll remove this and clean up the extra checks.